grpc/stream_test.go

```diff
@@ -0,0 +1,264 @@
+package grpc
+
+import (
+	"context"
+	"testing"
+
+	"github.com/dop251/goja"
+	"github.com/grafana/xk6-grpc/grpc/testutils/grpcservice"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.k6.io/k6/lib"
+	"go.k6.io/k6/metrics"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"gopkg.in/guregu/null.v3"
+)
+
+type FeatureExplorerStub struct {
+	grpcservice.UnimplementedFeatureExplorerServer
+
+	getFeature   func(ctx context.Context, point *grpcservice.Point) (*grpcservice.Feature, error)
+	listFeatures func(rect *grpcservice.Rectangle, stream grpcservice.FeatureExplorer_ListFeaturesServer) error
+}
+
+func (s *FeatureExplorerStub) GetFeature(ctx context.Context, point *grpcservice.Point) (*grpcservice.Feature, error) {
+	if s.getFeature != nil {
+		return s.getFeature(ctx, point)
+	}
+
+	return nil, status.Errorf(codes.Unimplemented, "method GetFeature not implemented")
+}
+
+func (s *FeatureExplorerStub) ListFeatures(rect *grpcservice.Rectangle, stream grpcservice.FeatureExplorer_ListFeaturesServer) error {
+	if s.listFeatures != nil {
+		return s.listFeatures(rect, stream)
+	}
+
+	return status.Errorf(codes.Unimplemented, "method ListFeatures not implemented")
+}
+
+func TestStream(t *testing.T) {
+	t.Parallel()
+
+	tests := []testcase{
+		{
+			name: "MethodNotFound",
+			initString: codeBlock{
+				code: `
+				var client = new grpc.Client();
+				client.load([], "../vendor/google.golang.org/grpc/test/grpc_testing/test.proto");`,
+			},
+			vuString: codeBlock{
+				code: `
+				client.connect("GRPCBIN_ADDR");
+				new grpc.Stream(client, "foo/bar")`,
+				err: `method "/foo/bar" not found in file descriptors`,
+			},
+		},
+		{
+			name: "InvalidParam",
```

Comment:
I would prefer to move this and the next cases in a specific unit test for the `newCallParams`. I think it will help the maintenance of this test in the long term and centralize the assertion of the shared params between invoke and stream.

Reply:
Yeah, I have the same feeling. And to be honest, generally don't like the current way of organizing the tests/subtests. It's really hard to read, maintain, and so on.